PhysicsTools/SelectorUtils/interface/VersionedIdProducer.h

```diff
@@ -102,16 +105,21 @@ VersionedIdProducer(const edm::ParameterSet& iConfig) {
     edm::LogWarning("IdInformation")
       << idmsg.str();
 
+    
+
     produces<std::string>(idname);
     produces<edm::ValueMap<bool> >(idname);
     produces<edm::ValueMap<float> >(idname); // for PAT
-    produces<edm::ValueMap<unsigned> >(idname);
+    produces<edm::ValueMap<unsigned> >(idname);  
+    produces<edm::ValueMap<unsigned> >(idname+std::string(bitmap_label));
```

Comment:
I'm using ValueMap in Tree::Draw quite extensively.
You don't need Ptr/Ref for it,
just need to have the ValueMap correspond only to one parent collection


On 5/19/15 6:03 AM, Lindsey Gray wrote:
> In PhysicsTools/SelectorUtils/interface/VersionedIdProducer.h
> <https://github.com/cms-sw/cmssw/pull/9002#discussion_r30588493>:
> 
>>      produces<std::string>(idname);
>>      produces<edm::ValueMap<bool> >(idname);
>>      produces<edm::ValueMap<float> >(idname); // for PAT
>> -    produces<edm::ValueMap<unsigned> >(idname);
>> +    produces<edm::ValueMap<unsigned> >(idname);  
>> +    produces<edm::ValueMap<unsigned> >(idname+std::string(bitmap_label));
> 
> Hi Guys, no one would ever use TTree::Draw() with an edm::ValueMap,
> since you cannot use edm::Ptr/Ref within TTree::Draw() to begin with
> (and the syntax would break the Draw syntax afaik).
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9002/files#r30588493>.
>